dockerfiles/cpp-23.Dockerfile

```diff
@@ -0,0 +1,36 @@
+FROM gcc:13.2.0-bookworm
+
+ENV CODECRAFTERS_DEPENDENCY_FILE_PATHS="vcpkg.json,vcpkg-configuration.json,CMakeLists.txt"
+
+RUN apt-get update && \
+    apt-get install --no-install-recommends -y zip=3.* && \ 
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN pwd && \
+    git clone https://github.com/microsoft/vcpkg.git && \
+    ./vcpkg/bootstrap-vcpkg.sh -disableMetrics
+
+RUN export "VCPKG_ROOT"="$(pwd)" && \
+    export PATH="$VCPKG_ROOT:$PATH"
+
+# cmake 3.29.2 is required by vcpkg
+RUN wget --progress=dot:giga https://github.com/Kitware/CMake/releases/download/v3.29.2/cmake-3.29.2-Linux-x86_64.tar.gz && \
+    tar -xzvf cmake-3.29.2-Linux-x86_64.tar.gz && \
+    mv cmake-3.29.2-linux-x86_64/ /cmake-portable
+
+WORKDIR /app
+
+COPY vcpkg.json ./
+COPY vcpkg-configuration.json ./
+COPY CMakeLists.txt ./
+COPY src ./src
```

Comment:
We definitely can't do this since it won't work with caching. We'll need to do something like create a fake `app.c` file if that's needed for building